src/ui/shared/routing/helpers.ts

```diff
@@ -0,0 +1,5 @@
+import { emitter } from '../events';
+
+export function navigateProgrammatically(params: { pathname: string }) {
+  emitter.emit('navigationRequest', params);
```

Comment:
why do we need this wrapper over `navigate`?

Reply:
because `navigate` works only in context of react router
we should migrate to creating a `router` instance later: https://reactrouter.com/en/main/upgrading/v6-data